File: DroidsAPI/src/main/java/com/github/rochedo098/droidsapi/api/script/ScriptFile.java
package com.github.rochedo098.droidsapi.api.script;

import com.github.rochedo098.droidsapi.DroidsAPI;
import dev.latvian.mods.rhino.Context;
import dev.latvian.mods.rhino.Scriptable;
import org.apache.commons.io.IOUtils;

import java.io.BufferedInputStream;
import java.io.IOException;
import java.io.InputStream;
import java.nio.charset.StandardCharsets;
import java.util.List;

import static com.github.rochedo098.droidsapi.api.script.ScriptManager.createStream;
import static com.github.rochedo098.droidsapi.api.script.ScriptManager.getPath;


public class ScriptFile {
    private final Context context = Context.enter();
    private final Scriptable scope = context.initStandardObjects();

    public String script;

    public ScriptFile(String script) {
        this.script = script;
    }

    public void load() {
        try(InputStream stream = createStream(getPath(this.script))) {
            String script2 = new String(IOUtils.toByteArray(new BufferedInputStream(stream)), StandardCharsets.UTF_8);
            context.evaluateString(scope, script2, getPath(script).toString(), 1, null);
        } catch (IOException e) {
            DroidsAPI.LOGGER.error(e);
        }
    }

    public void load_all() {
        List<String> files = ScriptManager.getScripts();
        for (String file : files) {
            new ScriptFile(file).load();
        }
    }
}

File: src/main/kotlin/com/github/rochedo098/droids/api/script/ScriptFile.kt
package com.github.rochedo098.droids.api.script

import com.github.rochedo098.droids.Droids
import dev.latvian.mods.rhino.Context
import dev.latvian.mods.rhino.Scriptable
import org.apache.commons.io.IOUtils
import java.io.BufferedInputStream
import java.io.IOException
import java.nio.charset.StandardCharsets

class ScriptFile(var script: String) {
    private val context: Context = Context.enter()
    private val scope: Scriptable = context.initStandardObjects()

    private fun load() {
        try {
            ScriptManager.createStream(
                ScriptManager.getPath(
                    script
                )
            ).use { stream ->
                val script2 = String(IOUtils.toByteArray(BufferedInputStream(stream)), StandardCharsets.UTF_8)
                context.evaluateString(
                    scope, script2, ScriptManager.getPath(
                        script
                    ).toString(), 1, null
                )
            }
        } catch (e: IOException) {
            Droids.LOGGER.error(e)
        }
    }

    private fun load_all() {
        val files: List<String> = ScriptManager.getScripts()
        for (file in files) {
            ScriptFile(file).load()
        }
    }
}